toolchain/sem_ir/inst_kind.h

```diff
@@ -92,8 +91,11 @@ static_assert(sizeof(InstKind) == 1, "Kind objects include padding!");
 // are not copyable, and only one instance of this type is expected to exist per
 // instruction kind, specifically `TypedInst::Kind`. Use `InstKind` instead as a
 // thin wrapper around an instruction kind index.
+template <typename TypedNodeArg>
 class InstKind::Definition : public InstKind {
  public:
+  using TypedNode = TypedNodeArg;
```

Comment:
Maybe an `Id` suffix would help readers know what this represents:
```suggestion
  using TypedNodeId = TypedNodeArg;
```

Reply:
Done